Remotion/Core/Core/Design/EnvDteDesignModeHelper.cs

```diff
@@ -66,27 +66,27 @@ public override System.Configuration.Configuration GetConfiguration ()
       }
     }
 
-    public object GetDesignTimePropertyValue (string propertyName)
+    public object? GetDesignTimePropertyValue (string propertyName)
     {
       ArgumentUtility.CheckNotNullOrEmpty ("propertyName", propertyName);
 
       try
       {
         //EnvDTE._DTE environment = (EnvDTE._DTE) ((IServiceProvider)site).GetService (typeof (EnvDTE._DTE));
-        Type _DTEType = TypeUtility.GetType ("EnvDTE._DTE, EnvDTE", true);
+        Type? _DTEType = TypeUtility.GetType ("EnvDTE._DTE, EnvDTE", true);
         object environment = DesignerHost.GetService (_DTEType);
 
         if (environment != null)
         {
           //EnvDTE.Project project = environment.ActiveDocument.ProjectItem.ContainingProject;
-          object activeDocument = _DTEType.InvokeMember ("ActiveDocument", BindingFlags.GetProperty, null, environment, null);
-          object projectItem = activeDocument.GetType().InvokeMember ("ProjectItem", BindingFlags.GetProperty, null, activeDocument, null);
-          object project = projectItem.GetType().InvokeMember ("ContainingProject", BindingFlags.GetProperty, null, projectItem, null);
+          object? activeDocument = _DTEType?.InvokeMember ("ActiveDocument", BindingFlags.GetProperty, null, environment, null);

```

Comment:
@iam-mholle that's a case for either null-agnostic code-section until officially cleaned up or the null-forgiving (damnit!) operator. The code expected this to be not-null and would not work if there was null. in the future, one could argue that this would be extended with assertion.isnotnull calls to make this 'type safe'.

Right now, we want to have the minimal possible change for code that ignores the null-option because there can never be null during normal operation in the first place. As a rule of thumb: for reflection cases such as these and any case where we control the entire scenario but the api supports null for other scenarios, we know that this can never be null unless something bad(TM) happens. In those cases, we would then use the asstion.isnotnull(...) check. 

On the other hand, cases where we simply forgot that null is a possible value, too, we found a bug that should then be fixed in a separate issue. you probably want to use the null-forgiving operator, too, but add an inline comment to a jira ticket that deals with fixing the null-bugs.